include/zephyr/devicetree.h

```diff
@@ -4172,6 +4229,16 @@
 #define DT_INST_NODE_HAS_PROP_AND_OR(inst, prop) \
 	DT_INST_NODE_HAS_PROP(inst, prop) ||
 
+/**
+ * @def DT_U64_C
+ * @brief Macro to add ULL postfix to the devicetree address constants
+ */
+#ifdef _LINKER
```

Comment:
```suggestion
#if defined(_LINKER) || defined(_ASMLANGUAGE)
```

I guess this needs to be like this if stdint.h can't be used in ASM contect